### PR TITLE
Site Editor: Fix the typo in the title label map

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR fixes the title label for templates in the Site Editor.

## Why?
It was incorrect.

## Testing Instructions
1. Open the Site Editor.
2. Confirm page title is rendered correctly.

## Screenshots or screencast 
| Before | After |
|--------|--------|
| CleanShot 2023-07-27 at 22 28 22 | CleanShot 2023-07-27 at 22 27 54 |

### CODE REVIEW

File: packages/edit-site/src/components/editor/index.js [hunk 1]
```diff
@@ -165,12 +165,11 @@ export default function Editor( { isLoading } ) {
 
 	let title;
 	if ( hasLoadedPost ) {
-		const type = typeLabels[ editedPostType ] ?? __( 'Template' );
 		title = sprintf(
 			// translators: A breadcrumb trail in browser tab. %1$s: title of template being edited, %2$s: type of template (Template or Template Part).
 			__( '%1$s ‹ %2$s ‹ Editor' ),
 			getTitle(),
-			type
+			typeLabels[ editedPostType ] ?? typeLabels.wp_template
```
Comment (Mamaduka) [inline packages/edit-site/src/components/editor/index.js:172]: Unrelated change; I hope it's okay to include it here. I just simplified things a bit.

**Review by richtabor — APPROVED** [review]
Nice catch.

File: packages/edit-site/src/components/editor/index.js (hunk 1 above)
Comment (aaronrobertshaw) [reply packages/edit-site/src/components/editor/index.js:172]: I don't have any issues with the change. It could be a personal preference for readability but simple is good.

**Review by aaronrobertshaw — APPROVED** [review]
LGTM as well

**Comment by ramonjd** [comment]
I just cherry-picked this PR to the update/packages-wp-6-3-RC3 branch to get it included in the next release: dacafc0f03
